src/DownloadManager.ts

```diff
@@ -0,0 +1,243 @@
+import { logger } from './Logger';
+
+import cliProgress from 'cli-progress';
+import WebSocket from 'ws';
+
+
+export class DownloadManager {
+    private webSocket: WebSocket;
+    // TODO: there's a "not a tty" mode for progresBar
+    // NOTE: is there a way to fix the ETA? Can't get size nor ETA from aria that I can see
+    // we initialize this for each download
+    private progresBar!: cliProgress.Bar;
+    private completed: number;
+    private queue: Set<string>;
+    private index: number;
+
+    public constructor(port: number) {
+        this.webSocket = new WebSocket(`http://localhost:${port}/jsonrpc`);
+        this.completed = 0;
+        this.queue = new Set<string>();
+        this.index = 1;
+
+        // Is this really needed having the 30 columns default if
```

Comment:
It doesn't have to be 30 but we need to hardcode something else we're right at the beginning where the progress library doesn't know the terminal width. Since we don't really support broken terminals 30 columns is bad enough for people to actually read the warning and use a proper terminal.

Reply:
I'll leave it be for now